### PR TITLE
Refactor tenant service

### PR DESCRIPTION
**Pull request by Pe5h4**
- implement new method `getTenantData` to obtain tenant data
- refactor naming for `get_current_tenant`, `set_tenants` and `_extract_tenant_from_url` from snake_case to camelCase format

### CODE REVIEW

**Comment by Pe5h4** [comment]
@ateska This is a PR about implementation `getTenantData` method within Tenant service (as we spoke of on review).

This PR is connected to other UI's (ASAB-Library-UI, BS-UI, LM.io UI) where schema is being obtained if specified in tenant's custom data. So this is just a part of smaller refactorisation, which is taking places in other UI's.

In following gif image is a bigger picture of usage.

- It displays LM.io UI, where Schema service from ASAB/Library is used in Discover and Analysis component. Integral part of ASAB Library Schema service is a usage of `getTenantData` method from Tenant service to obtain the Schema name. Method `getSchema` in ASAB Library Service is then used to get the particular schema of a tenant (if specified). (this is a par of ASAB Librayr UI MR on gitlab)

- The  `getSchema` method of ASAB Library Schema service is used in Discover and Analysis components to obtain the tenant related schema. (this is a part of LM.io UI and BS UI MR's on gitlab)

All changes has been made based on our discussion on friday's review.

tenant-service

File: src/modules/tenant/service.js [hunk 1]
```diff
@@ -67,19 +67,39 @@ export default class TenantService extends Service {
...
-	get_current_tenant() {
+	/*
+		getTenantData() method is used for returning the tenant data
+		It requires SeaCat Auth service
+	*/
+	async getTenantData() {
+		let tenantData = {};
+		let currentTenant = this.getCurrentTenant();
+		if (currentTenant) {
+			const SeaCatAuthAPI = this.App.axiosCreate('seacat_auth');
```
Comment (ateska) [inline src/modules/tenant/service.js:78]: This call can be cached, tenant data are quite static.
On the other hand, ETag-based caching can be quite OK here.

**Review by ateska — CHANGES_REQUESTED** [review]
Very small remark, not practically blocking a merge.

**Comment by Pe5h4** [comment]
@ateska Added cache to the tenant data: https://github.com/TeskaLabs/asab-webui/pull/388/commits/efb8e56d03cee6b7f5e5807390df60e97197ec30

File: src/modules/tenant/service.js (hunk 1 above)
Comment (Pe5h4) [reply src/modules/tenant/service.js:84]: Not sure if ETag would help here much, but I can take a look on that deeper if needed

**Comment by Pe5h4** [comment]
@ateska Can you please re-review?

File: src/modules/tenant/service.js (hunk 1 above)
Comment (ateska) [reply src/modules/tenant/service.js:84]: Please add "TODO" here: "Consider caching tenant data to reduce number of calls to the backend"

**Review by ateska — CHANGES_REQUESTED** [review]
Last remark, then mergeable,

**Review by ateska — APPROVED** [review]

File: src/modules/tenant/service.js (hunk 1 above)
Comment (Pe5h4) [reply src/modules/tenant/service.js:84]: @ateska tenant data are being cached as per refactorisation in this commit: https://github.com/TeskaLabs/asab-webui/commit/efb8e56d03cee6b7f5e5807390df60e97197ec30 , so I suppose there is no need to add that TODO, unless you wanted it for the `ETag`

**Pull request merged**